packages/proxy/lib/http/request-middleware.ts

```diff
@@ -305,11 +301,12 @@ const SendRequestOutgoing: RequestMiddleware = function () {
 
 export default {
   LogRequest,
+  CorrelateBrowserPreRequest,
```

Comment:
Does moving this up in the queue cause a performance impact? Since we asynchronously wait on the on CDP to send down the pre-request, this middleware will now block the following middleware while waiting on the CDP pre-request instead of allowing the other middlewares to complete.

Did that make any sense? 🤷‍♂️

Reply:
It does. I don't think it causes more of a performance issue than what already exists, or at least not significantly. I could see that delay happening earlier adding a few microseconds to maybe a 2/10 of a millisecond total to the middleware. I only say that because we know the jobs that happened before the `correlateBrowserPrerequest` take microseconds usually
<img width="1172" alt="Screenshot 2023-05-05 at 10 48 46 AM" src="https://user-images.githubusercontent.com/3980464/236491776-fd914fde-ecc4-4d74-84cc-90928847c9da.png">